gym_minigrid/envs/dynamic_minigrid.py

```diff
@@ -90,22 +132,43 @@ def alter_goal_pos():
                 self.goal_pos = new_goal_pos  # change the attribute
                 self.grid.set(*new_goal_pos, Goal())  # change the actual element in the grid
                 self.grid.set(*goal_pos, None)  # remove the previous goal
+                if not is_solvable():
+                    # revert the change
+                    self.goal_pos = goal_pos
+                    self.grid.set(*goal_pos, Goal())
+                    self.grid.set(*new_goal_pos, None)
+
+            if n_tries == n_tries_max:
+                raise Warning("Could not alter the agent start position "
+                              "in a reasonable amount of trials. Return without change")
 
         def set_or_remove_obj(obj):
 
-            while True:
+            solvable_and_changed = False
```

Comment:
why `solvable_and_changed`? suggests a difference to the other functions, is there one?